sdk/python/kfp/compiler/compiler.py

```diff
@@ -1037,8 +1037,13 @@ def compile(self,
 
     if self._mode == dsl.PipelineExecutionMode.V2_COMPATIBLE:
       pipeline_name = getattr(pipeline_func, '_component_human_name', '')
+      # pipeline names have one of the following formats:
+      # * pipeline/<name>
+      # * namespace/<ns>/pipeline/<name>
+      # when compiling, we will only have pipeline/<name>, but it will be overriden
+      # when uploading the pipeline to KFP API server.
       self._pipeline_name_param = dsl.PipelineParam(name='pipeline-name',
-                                                    value=pipeline_name)
+                                                    value=f'pipeline/{pipeline_name}')
```

Comment:
Thought: can this __{pipeline_name}__ contains `/`? Just thinking out loud that if our pipeline name formatting namespace/<ns>/pipeline/<name> is recognizable throughout the system workflow.

Reply:
It's specified in @pipeline annotations: https://github.com/kubeflow/pipelines/blob/f1f99c01e687be8f59564b7cc5649a053555dc77/samples/test/fail.py#L30

We do not set any restrictions for it in KFP. However, because namespace name must be a valid kubernetes resource name and cannot contain "/". We can at least explicitly find out the `namespace/<ns>/pipeline/` prefix without confusion. It doesn't matter whether pipeline name will have any special characters, that will just be after the `pipeline/` prefix.

Reply:
Sounds good, thank you Yuan for the explanation!